### DISCUSSION TITLE
What filters are being applied to a page?

### DISCUSSION THREAD

**Discussion by carolinaisslaying** [uBlockOrigin/uBlock-issues#3421] [Q&A]
Hi, just wondering if there is a way to view what filters are being applied to a site?

I've tried figuring it out using logger and public lists, but seemingly the rules I find by themselves do not actually block anything for the site, and ads come through regardless. 

(I am trying to figure out the exact rules that block ads on a site - ads are fully blocked normally but trying to test the filter rules I find against the site to figure out what rules actually block the ads).

Appreciate any help with this!

(The specific website is www.tvnz.co.nz - in terms of when watching videos, but I'm mainly asking if there is a specific tool in uBlock Origin that shows what filters are being applied to a site).

**Comment by gorhill** [comment]
The logger, see <https://github.com/gorhill/uBlock/wiki/The-logger>:

> uBlock Origin (uBO) comes with a logger, which gives the ability to inspect what uBO is doing with network requests and DOM elements, whether something is blocked or allowed, and which filter, if any, matched a network request or DOM element. This logger is unified, meaning it will display everything uBO does as it occurs.
> 
> The logger is the tool of choice to see, understand, diagnose and fix the functioning of uBO.

**Reply by garry-ut99** [reply]
Also additionally (just to make sure OP won't miss it):
- https://github.com/uBlockOrigin/uBlock-issues/issues/3224 ==> 
https://github.com/gorhill/uBlock/wiki/The-logger#dom-inspector ==> 
https://github.com/gorhill/uBlock/wiki/DOM-inspector

additionaly you may have want to:
- open DOM Inspector in a separate window ==> 
https://github.com/gorhill/uBlock/wiki/The-logger#tip (just like on the screenshoot) 
and then reduce its size (just like on the screenshoot)
